_posts/2017-08-02-July.md

```diff
@@ -0,0 +1,51 @@
+- Server - The sever works like it did in v0.2, hosting an Eve instance listening on a websocket. We also include a static file server that hosts the output of your program.
+
+- Browser Client - The browser client connects to a remote Eve server to which it streams records like key presses; and receives records like ticks of a timer. All the functionality from previous versions is still present -- no matter where Eve is running, the watcher system let's it interact with your browser (or anything else running JS or Rust) as if it were local.
+
+- Persistance - This is one of the features that has been missing since v0.2 and frequently requested. Persistence is implemented by writing to an append-only binary file at a user specified location. Right now this is a flag passed to the Eve executable, but this could be controlled from within Eve itself. Transactions are written to this log as they are executed in the runtime, and only those needed to reconstruct the state are persisted. We have no guarantees on durability/consistency yet, but that work will come when we start to tackle Eve in a distributed context.
+
+What about crashes?
+
+- Library - The entire standard library will be re-implemented in Rust for a number of reasons:
+
+  1. Performance
+  2. JS functions require a browser context, and therefore cannot be run in the terminal.
+  
+  This is largely a matter of wrapping native Rust functions for use in Eve, but we want to hold off on that until we implement a custom number system we have planned (more on that in a future post). We've added old watchers like `file`, `console`, and `compiler`; as well as new watchers like `http` and `json`. These newest additions allow you to parse JSON files into records, or send HTTP requests to API endpoints and parse the response.
+
+![Error](/images/v0.4-error.png)
+
+- Errors - We are reporting errors in the console, including line numbers and a helpful hint. We will also make these errors available through a request to the server, allowing 3rd party developer/editor extensions to integrate error reporting. That said, this work was mostly a matter of plumbing; providing good error messages is an ongoing process.
+
+- More testing - Rust has really nice testing facilities that make writing and running unit tests very easy. We've written a bunch of those that broadly cover the core functionality of the language. Testing in this context also means we just have to write a bunch of Eve applications to probe the runtime for weak spots.
+
+- Packaging and Distribution - To reach a wider audience, packaging and distribution will be a primary focus for this release. Packaging the Eve server is mostly handled for us by Rust, as it emits executables for all major platforms. The server can even host a native Eve editor. We've also experimented with embedding the Eve server in an Electron app. This solution allows us to present Eve as more of a native application, separate from the browser.
+
+## v0.4 timeline
```

Comment:
The order seems weird here. We talk about all the specifics of 0.4 before the general purpose of it.